src/maestro/agents/code_agent.py

```diff
@@ -18,6 +22,88 @@ def __init__(self, agent: dict) -> None:
         Initializes the agent with agent definitions.
         """
         super().__init__(agent)
+        self.agent = agent  # Store the agent dictionary for accessing metadata
+
+    def _install_dependencies(self) -> None:
+        """
+        Check if the agent has dependencies in its metadata and install them if they exist.
+        """
+        dependencies = self.agent.get("metadata", {}).get("dependencies")
+        if not dependencies or dependencies.strip() == "":
+            return
+
+        self.print(f"Installing dependencies for {self.agent_name}...")
+
+        temp_file_path = None
+        try:
+            # Create a temporary requirements.txt file
```

Comment:
I like the idea of this, but what was the motivation/intention of not just in-lining the dependencies in the command and instead putting them in a requirements file first?

Reply:
Added support pending ci